### PR TITLE
Add titlebar acrylic

### PR DESCRIPTION
**Pull request by matthew4850**
Add support for acrylic in the titlebar

## PR Checklist
* [x] CLA signed
## Detailed Description of the Pull Request / Additional comments
This seems to be a highly requested feature and seeing as #5772 was closed I thought it made sense to make a PR for this.
image

## Validation Steps Performed
Checked that acrylic works in both dark and light modes and switching between them still works. Also checked that acrylic in the tab row still works when tabs in titlebar is disabled.

### CODE REVIEW

**Comment by zadjii-msft** [comment]
Okay so I love this, and is almost certainly exactly how I'd implement it. 

I might have not been the most clear when I closed #5772. I do plan on coming back to XAML theming (#3327). It's my _personal_ passion project, probably one of the features that I'm most excited about. I unfortunately have other priorities that are more important right now (mainly #5000), so I'm pretty confident that #3327 isn't going to land in the 2.0 timeframe. I know the team wasn't going to have the time to come through and review that spec in any short order (it's already 12 months old haha). Plus, the spec needed a decent amount of additional work to reconcile it with [Mica](https://docs.microsoft.com/en-us/windows/apps/design/style/mica). I thought it best to close the spec PR for now and help clear out the PR backlog, and re-open when the team actually will have a chance to review. 

I don't want to be the kind of person who turns away valid and valuable community contributions in the name of "the big picture". I especially don't want to do that when I have no idea when the big picture will work itself out. So I'm gonna give this a think for a while to see if there's a reasonable way to accept this now and reconcile with themes later.

**Comment by matthew4850** [comment]
> Okay so I love this, and is almost certainly exactly how I'd implement it.

Thank you so much for this, I am new to C++/WinRT so it really means a lot.

> I might have not been the most clear when I closed #5772. I do plan on coming back to XAML theming (#3327). It's my _personal_ passion project, probably one of the features that I'm most excited about. I unfortunately have other priorities that are more important right now (mainly #5000), so I'm pretty confident that #3327 isn't going to land in the 2.0 timeframe. I know the team wasn't going to have the time to come through and review that spec in any short order (it's already 12 months old haha). Plus, the spec needed a decent amount of additional work to reconcile it with [Mica](https://docs.microsoft.com/en-us/windows/apps/design/style/mica). I thought it best to close the spec PR for now and help clear out the PR backlog, and re-open when the team actually will have a chance to review.
> 
> I don't want to be the kind of person who turns away valid and valuable community contributions in the name of "the big picture". I especially don't want to do that when I have no idea when the big picture will work itself out. So I'm gonna give this a think for a while to see if there's a reasonable way to accept this now and reconcile with themes later.

I hope it shouldn't be too hard to reconcile with themes, when necessary I think it should be possible to just remove this code and have some kind of auto-upgrade for the settings to something compliant with themes.

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -140,6 +140,44 @@ namespace winrt::TerminalApp::implementation
         }
         CATCH_LOG();
 
+        if (_settings.GlobalSettings().UseAcrylicInTabRow())
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TerminalPage.cpp:143]: Out of curiosity, will this setting hot-reload? (i.e. if the user changes the `settings.json` file while the Terminal is open, will the new value apply?)

I'd suspect that moving this to `TerminalPage::_RefreshUIForSettingsReload` might take care of that though

**Review by zadjii-msft — APPROVED** [review]
Okay, team consensus was that we're okay doing _this_ for now, and migrating it to themes when themes arrive in ~3.0. We're not going to take any of the other titlebar theming settings in 2.0. 

(We might take a fix for following the system setting for "use accent color in the titlebar", but with an opt-out setting (that again, would be migrated to themes in 3.0). The rest of the theming asks though will have to wait.)

I'm okay with
* this setting not hot-reloading right now. The team can always follow up and add that if needed
* the team can come through and link this up to the settings UI in a follow-up, no need to delay this PR any longer. 

Thanks again for the contribution here!

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 1 above)
Comment (matthew4850) [reply src/cascadia/TerminalApp/TerminalPage.cpp:143]: This won't hot-reload.
I'm guessing moving it to `TerminalPage::_RefreshUIForSettingsReload` would make hot-reloading work for enabling it but would require additional changes for disabling it.

Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp:143]: > but would require additional changes for disabling it.

that's a really good point. I'll say :shipit: for now and we can re-visit hot-reloading in a follow-up.

**Comment by matthew4850** [comment]
> the team can come through and link this up to the settings UI in a follow-up, no need to delay this PR any longer.

I might be misunderstanding but the PR already contains a setting UI toggle.

File: src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw [hunk 2]
```diff
@@ -343,6 +343,14 @@
     <value>When unchecked, the title bar will appear above the tabs.</value>
     <comment>A description for what the "show titlebar" setting does. Presented near "Globals_ShowTitlebar.Header".</comment>
   </data>
+  <data name="Globals_AcrylicTabRow.Header" xml:space="preserve">
+    <value>Show acrylic in tab row (requires relaunch)</value>
+    <comment>Header for a control to toggle whether acrylic shows in the tab row. Changing this setting requires the user to relaunch the app.</comment>
+  </data>
+  <data name="Globals_AcrylicTabRow.HelpText" xml:space="preserve">
+    <value>When checked, the tab row will have the acrylic material.</value>
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw:351]:
```suggestion
    <value>When enabled, the tab row will have the acrylic material.</value>
```

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
This looks great! A few things before I sign off:
- Please add `useAcrylicInTabRow` to the schema
- Please add this to the docs repo
- Please file a follow-up issue for this to be hot-reloadable

**Comment by zadjii-msft** [comment]
@carlos-zamora Mind if we do those in post? Not sure I want to necessarily force an external contributor to comply with the fullness of our bureaucracy 😜

**Review by carlos-zamora — APPROVED** [review]

**Comment by zadjii-msft** [comment]
Ah shoot, there are merge conflicts though - @matthew4850 if you want to fix those, otherwise I might just try on monday

**Comment by matthew4850** [comment]
@carlos-zamora apologise for not getting round to your requested changes, I was away and only just got back.

**Comment by matthew4850** [comment]
> Ah shoot, there are merge conflicts though - @matthew4850 if you want to fix those, otherwise I might just try on monday

Resolved 🙂

**Comment by carlos-zamora** [comment]
> @carlos-zamora apologise for not getting round to your requested changes, I was away and only just got back.

No worries. Thank YOU for doing this! 🙂

**Pull request merged**